### PR TITLE
Update WSClient.java

### PR DESCRIPTION
**Pull request by BigBeaule**
Attempt to fix issue #38 in order to reuse the SSLSocketFactory

### CODE REVIEW

**Comment by michaelrice** [comment]
Thanks @BigBeaule for the p/r

It seems to be missing the import for SSLSocketFactory which is why travis-ci is reporting the [failure](https://travis-ci.org/yavijava/yavijava/jobs/70095246#L124).

Normally its best practice to fork then create a topic/feature/bug fix branch, then make your changes to that branch and submit the p/r from that branch, but Im happy to take any help I can get :)

Would you mind adding that import, then amending your last commit to include the import so that travis-ci can run the tests? When you amend your commit you will need to `git push -f` to get it to go, and the pull request will automatically update and travis-ci will automatically run once that happens.

**Comment by michaelrice** [comment]
Hi @BigBeaule Thanks for fixing this up. Im going to pull it locally to rebase the extra commits out and such unless you want to take a stab at that. I also made this wiki page to help new folks out so your contributions are easy to consume into the project: https://github.com/yavijava/yavijava/wiki/Contributing

File: src/main/java/com/vmware/vim25/ws/WSClient.java [hunk 1]
```diff
@@ -68,7 +73,18 @@ public WSClient(String serverUrl, boolean ignoreCert) throws MalformedURLExcepti
...
         log.trace("Ignore ssl: " + ignoreCert);
         this.baseUrl = new URL(serverUrl);
-        this.ignoreCert = ignoreCert;
+        
+        try {
+            this.sslSocketFactory = ignoreCert ? TrustAllSSL.getTrustContext().getSocketFactory() : null;
+        } 
+        catch (NoSuchAlgorithmException e) {
+            log.debug("Unable to find suitable algorithm while attempting to communicate with remote server.", e);
+            throw new IllegalArgumentException("Unable to find suitable algorithm while attempting to communicate with remote server.", e);
```
Comment (aaron-brown) [inline src/main/java/com/vmware/vim25/ws/WSClient.java:82]: Note: Adding this means that you should add `IllegalArgumentException` to the `throws` clause, as a best-practice measure.

Not doing so will throw an `UndeclaredThrowableException`.

**Comment by michaelrice** [comment]
Ive pulled this down, rebased it down to 1 commit, made the changes @aaron-brown mentioned. Ive also opened it as a new p/r which we can use for the rest of this review. Please see #114

**Pull request closed without merging**